src/Evolution/Systems/ScalarTensor/Characteristics.hpp

```diff
@@ -0,0 +1,66 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include <cstddef>
+
+#include "DataStructures/DataBox/Tag.hpp"
+#include "DataStructures/DataVector.hpp"
+#include "DataStructures/Tensor/EagerMath/Magnitude.hpp"
+#include "DataStructures/Tensor/Tensor.hpp"
+#include "Evolution/Systems/CurvedScalarWave/Characteristics.hpp"
+#include "Evolution/Systems/GeneralizedHarmonic/Characteristics.hpp"
+#include "PointwiseFunctions/GeneralRelativity/Tags.hpp"
+#include "Utilities/Gsl.hpp"
+#include "Utilities/TMPL.hpp"
+
+namespace ScalarTensor::Tags {
+struct LargestCharacteristicSpeed : db::SimpleTag {
+  using type = double;
+};
+
+/*!
+ * \brief Computes the largest magnitude of the characteristic speeds.
+ *
+ * \details Compute the maximum of the largest characteristic speed of each
+ * component system, i.e. the largest speed between ::gh and ::CurvedScalarWave.
+ */
+template <typename Frame = Frame::Inertial>
+struct ComputeLargestCharacteristicSpeed : db::ComputeTag,
+                                           LargestCharacteristicSpeed {
+  static constexpr size_t Dim = 3_st;
+  using argument_tags = tmpl::remove_duplicates<
+      tmpl::append<typename gh::Tags::ComputeLargestCharacteristicSpeed<
+                       Dim, Frame>::argument_tags,
+                   typename CurvedScalarWave::Tags::
+                       ComputeLargestCharacteristicSpeed<Dim>::argument_tags>>;
```

Comment:
Just write the list out.  It's not obvious what this expression gives, but we know the required answer because it has to match the function arguments below.

Reply:
Wrote out the list.